File: October 2024 GFG SOLUTION/October-12.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends

1)
class Solution {
  public:
    int pairWithMaxSum(const vector<int>& arr) {
        int n = arr.size();

        if (n < 2) return -1;

        int maxSum = arr[0] + arr[1];  

        for (int i = 1; i < n - 1; ++i) {
            maxSum = max(maxSum, arr[i] + arr[i + 1]);
        }

        return maxSum;
    }
};

2)
class Solution {
  public:
    int pairWithMaxSum(vector<int>& arr) {
        int n = arr.size();
        if (n < 2)
            return -1;
        int maxSum = arr[0] + arr[1]; 
        for (int i = 1; i < n - 1; ++i) {
            int currentSum = arr[i] + arr[i + 1];
            maxSum = max(maxSum, currentSum); 
        }

        return maxSum;
    }
};

3)
class Solution {
  public:
    int pairWithMaxSum(vector<int>& arr) {
        if (arr.size() < 2)
            return -1;
        
        int maxSum = arr[0] + arr[1];
        for (int i = 1; i < arr.size() - 1; ++i)
            maxSum = max(maxSum, arr[i] + arr[i + 1]);

        return maxSum;
    }
};


//{ Driver Code Starts.

int main() {
    int tc;
    cin >> tc;
    cin.ignore();

    while (tc--) {
        string input;
        getline(cin, input);

        stringstream ss(input);
        vector<int> a;
        int number;
        while (ss >> number) {
            a.push_back(number);
        }

        Solution obj;
        int res = obj.pairWithMaxSum(a);
        cout << res << endl;
    }
}
// } Driver Code Ends

File: October 2024 GFG SOLUTION/October-12.py
1)
class Solution:
    def pairWithMaxSum(self, arr):
        n = len(arr)

        if n < 2:
            return -1

        max_sum = arr[0] + arr[1]  

        for i in range(1, n - 1):
            max_sum = max(max_sum, arr[i] + arr[i + 1])

        return max_sum

2)
class Solution:
    def pairWithMaxSum(self, arr):
        n = len(arr)
        if n < 2:
            return -1  

        max_sum = arr[0] + arr[1] 
        for i in range(1, n - 1):
            current_sum = arr[i] + arr[i + 1] 
            max_sum = max(max_sum, current_sum) 

        return max_sum

3)
class Solution:
    def pairWithMaxSum(self, arr):
        if len(arr) < 2:
            return -1 
        
        return max(arr[i] + arr[i + 1] for i in range(len(arr) - 1))

#{ 
 # Driver Code Starts
if __name__ == "__main__":
    import sys
    input = sys.stdin.read
    data = input().strip().split("\n")

    t = int(data[0])
    lines = data[1:]

    for line in lines:
        s = list(map(int, line.strip().split()))
        solution = Solution()
        res = solution.pairWithMaxSum(s)
        print(res)

# } Driver Code Ends
